src/main/java/Duke.java

```diff
@@ -1,10 +1,194 @@
+                                    throw new DukeException("☹ OOPS!!! The deadline must be completed "
+                                            + "by a certain date");
+                                } else {
+                                    try {
+                                        SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yyyy HHmm");
+                                        tasks.addItemToList(new Deadline(cmd.getInstruction(),
+                                                formatter.parse(cmd.getDate())));
+                                    } catch (java.text.ParseException exp) {
+                                        exp.printStackTrace();
+                                        break;
+                                    }
+                                    try {
+                                        storage.append(tasks.getLastItem());
+                                    } catch (Exception exp) {
+                                        ui.showErrorMsg(exp.getMessage());
+                                    }
+                                }
+                            } catch (DukeException exp) {
+                                ui.showErrorMsg(exp.getMessage());
+                            }
+                        } else if (cmd.getCommandType().equals("todo")) {
+                            try {
+                                if (cmd.getInstruction().equals("") || cmd.getInstruction().equals("todo")) {
+                                    throw new DukeException("☹ OOPS!!! The description of todo cannot be empty");
+                                } else {
+                                    tasks.addItemToList(new Todo(cmd.getInstruction()));
+                                    try {
+                                        storage.append(tasks.getLastItem());
+                                    } catch (Exception exp) {
+                                        ui.showErrorMsg(exp.getMessage());
+                                    }
+                                }
+                            } catch (DukeException exp) {
+                                ui.showErrorMsg(exp.getMessage());
+                            }
+                        } else if (cmd.getCommandType().equals("event") || cmd.getInstruction().equals("event")) {
+                            try {
+                                if (cmd.getInstruction().equals("")) {
+                                    throw new DukeException("☹ OOPS!!! The description of event cannot be empty");
+                                } else if (!echo.contains("/at")) {
+                                    throw new DukeException("☹ OOPS!!! The event must be at by a certain date");
+                                } else {
+                                    try {
+                                        SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yyyy HHmm");
```

Comment:
Maybe you can consider shifting more logic related to creating and adding tasks to addItemToList() method of TaskList, this can help to shorten the length of your run function.